### PR TITLE
Debug/Toolbar - Memory issue fix

### PR DESCRIPTION
**Pull request by najdanovicivan**
**Description**
This is the very simple method to avoid memory leak issue #4137 when entity objects with circular references are passed to the view. 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Debug/Toolbar.php [hunk 1]
```diff
@@ -468,4 +468,27 @@ protected function format(string $data, string $format = 'html'): string
...
 	}
+
+	/**
+	 * Process the varable to string for display
+	 *
+	 * @param mixed $var Variable
+	 *
+	 * @return boolean|string
+	 */
+	protected function processVar($var)
```
Comment (MGatner) [inline system/Debug/Toolbar.php:479]: Let's make this private for now, unless there are compelling reasons to allow access to this method.

File: system/Debug/Toolbar.php [hunk 2]
```diff
@@ -468,4 +468,27 @@ protected function format(string $data, string $format = 'html'): string
...
+	{
+		if (is_object($var))
+		{
+			$var = is_callable([$var, 'toArray']) ? $var->toArray() : get_class($var);
```
Comment (MGatner) [inline system/Debug/Toolbar.php:483]: This no longer a) tells us the object class for array-ables nor b) allows displaying valid, non-recursive objects that are not array-able.

**Review by MGatner — CHANGES_REQUESTED** [review]
I like the concept but not the end result. We need to find some way to catch as much as we can. Does Kint (already included) not have some solution for this in place?

**Comment by najdanovicivan** [comment]
@MGatner  I managed to make Kint work inside debug bar. The issue I had is that Kint render will return JS and CSS in as the string and since those are set as text via JS. The javascript from Kint will never be executed. So I had to play a bit with it to add the JS and CSS into the page head instead and then drop those when it's being rendered in the debug bar

**Comment by MGatner** [comment]
I'll take a look - if it is working then that's probably good. Kint also supports prefix modifiers to adjust the behavior - I think we could probably use `~` for "text output", or maybe `@` to "return output". Play around with those and see what you think. 

https://kint-php.github.io/kint/advanced/

File: system/Debug/Toolbar.php [hunk 3]
```diff
@@ -343,12 +363,19 @@ public function prepare(RequestInterface $request = null, ResponseInterface $res
 				return;
 			}
 
+			$oldKintMode        = Kint::$mode_default;
+			Kint::$mode_default = Kint::MODE_RICH;
+			$kintScirpt         = @Kint::dump('');
+			Kint::$mode_default = $oldKintMode;
+			$kintScirpt         = substr($kintScirpt, 0, strpos($kintScirpt, '</style>') + 8 );
```
Comment (MGatner) [inline system/Debug/Toolbar.php:370]:
```suggestion
			$kintScript         = substr($kintScript, 0, strpos($kintScirpt, '</style>') + 8 );
```
Typo in the variable name.

File: system/Debug/Toolbar.php [hunk 4]
```diff
@@ -343,12 +363,19 @@ public function prepare(RequestInterface $request = null, ResponseInterface $res
 				return;
 			}
 
+			$oldKintMode        = Kint::$mode_default;
+			Kint::$mode_default = Kint::MODE_RICH;
+			$kintScirpt         = @Kint::dump('');
```
Comment (MGatner) [inline system/Debug/Toolbar.php:368]: Ah, I see you already found the `@`.

File: system/Debug/Toolbar.php [hunk 5]
```diff
@@ -343,12 +363,19 @@ public function prepare(RequestInterface $request = null, ResponseInterface $res
...
+			Kint::$mode_default = $oldKintMode;
+			$kintScirpt         = substr($kintScirpt, 0, strpos($kintScirpt, '</style>') + 8 );
+
 			$script = PHP_EOL
 					. '<script type="text/javascript" {csp-script-nonce} id="debugbar_loader" '
 					. 'data-time="' . $time . '" '
 					. 'src="' . site_url() . '?debugbar"></script>'
 					. '<script type="text/javascript" {csp-script-nonce} id="debugbar_dynamic_script"></script>'
 					. '<style type="text/css" {csp-style-nonce} id="debugbar_dynamic_style"></style>'
+					. $kintScirpt
```
Comment (MGatner) [inline system/Debug/Toolbar.php:378]:
```suggestion
					. $kintScript
```

**Review by MGatner — CHANGES_REQUESTED** [review]
I think this should be fine. It definitely adds some complexity. See what you think of the text output option and if that is a worthy trade-off.

**Comment by najdanovicivan** [comment]
I am using @ prefix to return output. I tried text output as well but it looks nasty.  

I was unable to make it work by chaining prefixes eg @~dd() will not work for me. 

Another problem with text output is it can get very long especially when controller is being passed to the view and if any of the variables contains HTML it may break the <pre> tag

**Review by MGatner — APPROVED** [review]

**Pull request merged**